assets/src/components/progressBar/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Progress bar for displaying visual feedback about steps-completed.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import './style.scss';
+
+/**
+ * Progress bar.
+ */
+class ProgressBar extends Component {
+
+	/**
+	 * Get completion as a percentage.
+	 *
+	 * @return int
+	 */
+	getCompletionPercentage() {
+		let { completed, total } = this.props;
+		completed = parseInt( completed );
+		total = parseInt( total );
+
+		if ( ! total ) {
+			return 100;
+		}
+
+		return Math.min( Math.round( ( completed / total ) * 100 ), 100 );
+	}
+
+	/**
+	 * Render.
+	 */
+	render() {
+		const { label, completed, total, displayFraction } = this.props;
+
```

Comment:
Suggestion: sanitize the completed value to check that it is a number and is between 0-`total`, e.g.:

`const cleanCompleted = Math.max( 0, Math.min( ( parseInt( completed ) || 0 ), parseInt( total ) ) );`

In my testing it's possible to set the `completed` prop to a number greater than total, a number less than 0, or a string that yields `NaN` from `parseInt()`.